src/core/dash_internal.h

```diff
@@ -1636,11 +1621,8 @@ bool Segment<Key, Value, Policy>::TraverseLogicalBucket(uint8_t bid, HashFn&& hf
 
 template <typename Key, typename Value, typename Policy>
 size_t Segment<Key, Value, Policy>::SlowSize() const {
-  size_t res = 0;
-  for (unsigned i = 0; i < kTotalBuckets; ++i) {
-    res += bucket_[i].Size();
-  }
-  return res;
+  return std::accumulate(&bucket_[0], &bucket_[0] + kTotalBuckets, size_t(0),
```

Comment:
I would like to revert this change :) just because `std::acccumulate` exists, it does not mean using it makes the code more readable

Reply:
I agree, accumulate has some "mental" reading overhead... Really anticipate ranges until it becomes view(bucket_, bucket_ + offset).map(size).sum()

Reply:
IMO in this particular case I think accumulate is fine but I am more than happy to revert! @dranikpg I wanted to make the exact same comment but you got it first. Note that:

1. ranges::accumulate is not part of c++20 unfortunately but I think it's available in c++23 as `left_fold`. And you can replace the custom lambda with a `projection` (similar to `.map` you wrote)
2. boost::accumulate would suffice here as well

Generally, my main complain against STL algorithms (as with most of the world) is that you have to pass the range explicitly. I would personally opt in (as suggested in the past as well) to use one of the two libraries in our codebase and make our life a little easier with using those algorithms.